components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/util/SOAPOperationBindingUtils.java

```diff
@@ -0,0 +1,206 @@
+            try {
+                tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                        .getTenantId(tenantDomain);
+                registry = registryService.getGovernanceSystemRegistry(tenantId);
+                String resourcePath = APIConstants.API_LOCATION + RegistryConstants.PATH_SEPARATOR +
+                        provider + RegistryConstants.PATH_SEPARATOR + name + RegistryConstants.PATH_SEPARATOR + version
+                        + RegistryConstants.PATH_SEPARATOR + SOAPToRESTConstants.SOAP_TO_REST_RESOURCE;
+                return registry.resourceExists(resourcePath);
+            } catch (RegistryException e) {
+                handleException("Error when create registry instance ", e);
+            } catch (UserStoreException e) {
+                handleException("Error while reading tenant information ", e);
+            }
+        } finally {
+            if (isTenantFlowStarted) {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        }
+        return false;
+    }
+
+    /**
+     * gets parameters from the soap operation and populates them in {@link WSDLSOAPOperation}
+     *
+     * @param soapOperations soap binding operations
+     */
+    private static void populateSoapOperationParameters(Set<WSDLSOAPOperation> soapOperations) {
+        String[] primitiveTypes = { "string", "byte", "short", "int", "long", "float", "double", "boolean" };
+        List primitiveTypeList = Arrays.asList(primitiveTypes);
+        for (WSDLSOAPOperation op : soapOperations) {
+            String resourcePath;
+            String operationName = op.getName();
+            op.setSoapBindingOpName(operationName);
+            if (operationName.toLowerCase().startsWith("get")) {
+                resourcePath = operationName.substring(3, operationName.length());
+                op.setHttpVerb(HTTPConstants.HTTP_METHOD_GET);
+            } else {
+                resourcePath = operationName;
+                op.setHttpVerb(HTTPConstants.HTTP_METHOD_POST);
+            }
+            resourcePath = resourcePath.substring(0, 1).toLowerCase() +
+                    resourcePath.substring(1, resourcePath.length());
+            op.setName(resourcePath);
+
+            List<WSDLOperationParam> params = op.getParameters();
+            for (WSDLOperationParam param : params) {
+                if (param.getDataType() != null) {
+                    String dataTypeWithNS = param.getDataType();
+                    String dataType = dataTypeWithNS.substring(dataTypeWithNS.indexOf(":") + 1);
+                    param.setDataType(dataType);
+                    if (!primitiveTypeList.contains(dataType)) {
+                        param.setComplexType(true);
+                    }
+                }
+            }
+        }
+    }
+
+    public static WSDLSOAPOperationExtractor getWSDLProcessor(byte[] content, APIMWSDLReader wsdlReader) throws APIManagementException {
```

Comment:
Seems like the line exceeds 120 column width.

Reply:
Also, shall we add javadoc comments?

Reply:
fixed in e18f43f5dee9ed43a3239af62f38d67c72c78fa2